src/Illuminate/Console/Parser.php

```diff
@@ -105,11 +105,11 @@ protected static function parseOption($token)
 
         $matches = preg_split('/\s*\|\s*/', $token, 2);
 
+        $shortcut = null;
```

Comment:
Why would you set `$shortcut` twice ?

Reply:
> Why would you set `$shortcut` twice ?

### Before:

```php
if (isset($matches[1])) {
            $shortcut = $matches[0];
            $token = $matches[1];
        } else {
            $shortcut = null;
        }
```

And since the variable `$shortcut` is used later, it must have an initial value. Now we have the option to remove the else statement and have `$shortcut` initialized with a `Null` value.

### After:

```php
$shortcut = null;

        if (isset($matches[1])) {
            $shortcut = $matches[0];
            $token = $matches[1];
        }
```

**In fact, our goal with this change has been to keep the code simple.**

Reply:
Why would you now set its value twice?
If `$matches[1]` is set, you set `$shortcut` twice, once on line 108, then at line 111.
A `else` is not a complicated code statement, and this change does not add any value.

cc @driesvints

Reply:
### For example:

```php
/**
     * Fail the job from the queue.
     *
     * @param  \Throwable|string|null  $exception
     * @return void
     */
    public function fail($exception = null)
    {
        if (is_string($exception)) {
            $exception = new ManuallyFailedException($exception);
        }

        if ($exception instanceof Throwable || is_null($exception)) {
            if ($this->job) {
                return $this->job->fail($exception);
            }
        } else {
            throw new InvalidArgumentException('The fail method requires a string or an instance of Throwable.');
        }
    }
```

In your opinion, for example, why is the value of $exception being set again in the fail method of InteractWithQueue if it is of type string?

And if we search for such cases in Laravel, there are many of them. With all due respect, I don't find your reasoning convincing.


@lucasmichot